src/main/rules/GCI95/csharp/GCI95.asciidoc

```diff
@@ -0,0 +1,45 @@
+Using the `as` operator followed by a null check is more verbose and can be simplified using the `is` operator with pattern matching.
+
+When using the `is` pattern, the type check and cast happen in one concise statement, improving readability and reducing boilerplate.
+
+Therefore, using `is` with pattern matching will simplify code and improve maintainability.
+
+== Non compliant Code Example
+
+[source,csharp]
+----
+var myObj = obj as MyType;
+if (myObj != null)
+{
+    myObj.DoSomething();
+}
+----
+
+== Compliant Code Example
+
+[source,csharp]
+----
+if (obj is MyType myObj)
+{
+    myObj.DoSomething();
+}
+----
+
+[source,csharp]
+----
+var anotherObj = maybeObj as AnotherType;
+if (anotherObj != null)
+{
+    Console.WriteLine(anotherObj.Value);
+}
+----
+
+== Compliant Code Example
+
+[source,csharp]
+----
+if (maybeObj is AnotherType anotherObj)
+{
+    Console.WriteLine(anotherObj.Value);
+}
+----
```

Comment:
Could you please add some references to prove this rule ? web pages ou local performance tests ?